.github/workflows/actions.yml

```diff
@@ -233,23 +238,24 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: set sc version and artifact filename
-        run: |
-          # set SC_VERSION to tag (if present) or to commit SHA
-          SC_VERSION=
-          if [[ "$GITHUB_REF" == refs/tags/* ]]; then 
-            FULL_TAG=${GITHUB_REF#refs/tags/}
-            TAG=${FULL_TAG##Version-}
-            SC_VERSION=$TAG
-          else
-            SC_VERSION=$GITHUB_SHA
-          fi
-          echo "ARTIFACT_FILE=SuperCollider-$SC_VERSION-${{ matrix.artifact-suffix }}.zip" >> $GITHUB_ENV
+      - name: prepare daily timestamp for cache
+        id: current-date
+        run: echo "::set-output name=stamp::$(date '+%Y-%m-%d')"
+      - name: prepare monthly timestamp for cache
+        id: current-month
+        run: echo "::set-output name=stamp::$(date '+%Y-%m')"
       - name: cache ccache
         uses: actions/cache@v2
         with:
           path: ~/Library/Caches/ccache
-          key: ${{ runner.os }}-v1-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-${{ matrix.qt-version }} # -v2- etc can be used for "clearing" the cache, it's not functionally connected to the version of the script
+          key: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-${{ matrix.qt-version }}-${{ steps.current-date.outputs.stamp }}
+          restore-keys: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-${{ matrix.qt-version }}-
+      - name: cache homebrew downloads
+        uses: actions/cache@v2
+        id: cache-homebrew
+        with:
+          path: ~/Library/Caches/Homebrew/downloads
+          key: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.xcode-version }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-${{ matrix.qt-version }}-homebrew-${{ steps.current-month.outputs.stamp }}
```

Comment:
just wondering, why monthly instead of daily for homebrew? if there is an update in a big package i wouldn't want to redownload it on every build for so long; does it take longer to update the cache?

Reply:
We currently don't update homebrew on every run here; package versions are pretty current (eg. Qt is 5.15.1). I assume the images get periodically updated (though I haven't confirmed that) and I don't think this happens all that often. But if you have a strong feeling about this I can change this to daily.